core/src/main/scala/com/typesafe/tools/mima/core/Problems.scala

```diff
@@ -102,6 +102,18 @@ case class IncompatibleResultTypeProblem(oldmeth: MemberInfo, newmeth: MemberInf
   }
 }
 
+/**
+ * Produced when the basic types are the same, but the full signature is still different,
+ * for example when generic parameters don't match.
+ * This has a chance of false positives.
```

Comment:
Such as?

Or do you mean false negatives?  i.e. MiMa fails to see the change from `(A, B)` to `(B, A)`?

Reply:
From memory, I think it will produce spurious warnings in these cases:
* when the code changes `def foo[A]: Option[A]` to `def foo[B]: Option[B]`
* when it changes from `foo: Option[A]` to `foo: Option[B]` even if B is a subtype of A.

Reply:
Aren't those all `Lscala/Option<Ljava.lang.Object;>;`?

Maybe these should be test cases too, with this patch applied.

Reply:
OK, I'll add them.

Reply:
Thanks.  TIL some more.